drivers/sensor/maxim/ds18b20/ds18b20.c

```diff
@@ -25,15 +27,32 @@ LOG_MODULE_REGISTER(DS18B20, CONFIG_SENSOR_LOG_LEVEL);
 static int ds18b20_configure(const struct device *dev);
 
 /* measure wait time for 9-bit, 10-bit, 11-bit, 12-bit resolution respectively */
-static const uint16_t measure_wait_ms[4] = { 94, 188, 376, 750 };
+static const uint16_t measure_wait_ds18b20_ms[4] = { 94, 188, 376, 750 };
 
-static inline void ds18b20_temperature_from_raw(uint8_t *temp_raw,
-						struct sensor_value *val)
+/* ds18s20 always needs 750ms */
+static const uint16_t measure_wait_ds18s20_ms = { 750 };
+
+static bool is_ds18s20(const struct device *dev)
 {
-	int16_t temp = sys_get_le16(temp_raw);
+	const struct ds18b20_config *cfg = dev->config;
 
-	val->val1 = temp / 16;
-	val->val2 = (temp % 16) * 1000000 / 16;
+	return cfg->chip == type_ds18s20;
```

Comment:
Not sure this function really brings a benefit, it only makes a simple check.
I think is would be more clear to just do that check in the if conditions directly.